AndroidSDK/src/com/leanplum/LeanplumPushService.java

```diff
@@ -383,7 +383,33 @@ private static void showNotification(Context context, Bundle message) {
     }
 
     try {
-      notificationManager.notify(notificationId, builder.build());
+      // Check if we have a chained message, and if it exists in var cache.
```

Comment:
could you please create a method for this?

Reply:
Where should it go?  Same file?

Reply:
ActionContext.java ?

Reply:
The logic here works on `Bundle` and ActionContext.java deals with `Map`.  Would that still be appropriate?

Reply:
I extracted teh logic that works on just the action map to ActionContext.

Reply:
you can pass String  to this method - String actionData = message.getString(Constants.Keys.PUSH_MESSAGE_ACTION) and return true/false

Reply:
fixed

Reply:
fixed